Awful.apk/src/main/java/com/ferg/awfulapp/AwfulActivity.kt

```diff
@@ -73,8 +73,11 @@ abstract class AwfulActivity : AppCompatActivity(), AwfulPreferences.AwfulPrefer
                 if (ignoreFormkey == null || userAvatarUrl == null) RefreshUserProfileRequest(this@AwfulActivity).sendBlind()
                 if (ignoreFormkey == null) FeatureRequest(this@AwfulActivity).sendBlind()
             }
-        // TODO: this interferes with the code in AwfulFragment#reAuthenticate - i.e. it forces "return to this activity" behaviour, but fragments may want to return to the main activity.
-        // And the activity isn't always the authority, e.g. using BasicActivity which is a plain container where all the specific behaviour is handled by its fragment. It's awkward
+
+            CaptchaActivity.isCaptchaBeingHandled() -> { /* do nothing */ }
```

Comment:
I didn't quite understand what this is for. Can you explain?

Reply:
If the user is not logged in, without this line, the following lines will forcefully kill the captcha activity and open the login screen - making it impossible to log in (as solving the captcha is required for that).

I can add a comment explaining this, if you'd like!